main/inc/lib/webservices/Rest.php

```diff
@@ -1590,4 +1594,332 @@ private function encodeParams(array $additionalParams = [])
+                case 'official_code':
+                    $user->setOfficialCode($v);
+                    break;
+                case 'picture_uri':
+                    $user->setPictureUri($v);
+                    break;
+                case 'creator_id':
+                    $user->setCreatorId($v);
+                    break;
+                case 'competences':
+                    $user->setCompetences($v);
+                    break;
+                case 'diplomas':
+                    $user->setDiplomas($v);
+                    break;
+                case 'openarea':
+                    $user->setOpenArea($v);
+                    break;
+                case 'teach':
+                    $user->setTeach($v);
+                    break;
+                case 'productions':
+                    $user->setProductions($v);
+                    break;
+                case 'language':
+                    $languages = api_get_languages();
+                    if (!in_array($v, $languages['folder'])) {
+                        throw new Exception(get_lang('LanguageUnavailable'));
+                    }
+                    $user->setLanguage($v);
+                    break;
+                case 'registration_date':
+                    $user->setRegistrationDate($v);
+                    break;
+                case 'expiration_date':
+                    $user->setExpirationDate(new DateTime(
+                        api_get_utc_datetime($v),
+                        new DateTimeZone('UTC')
+                    ));
+                    break;
+                case 'active':
+                    // see UserManager::update_user() usermanager.lib.php:1205
+                    if ($user->getActive() != $v) {
+                        $user->setActive($v);
+                        Event::addEvent($v ? LOG_USER_ENABLE : LOG_USER_DISABLE, LOG_USER_ID, $userId);
+                    }
+                    break;
+                case 'openid':
+                    $user->setOpenId($v);
+                    break;
+                case 'theme':
+                    $user->setTheme($v);
+                    break;
+                case 'hr_dept_id':
+                    $user->setHrDeptId($v);
+                    break;
+                case 'extra':
+                    if (is_array($v)) {
+                        if (count($v) > 0) {
+                            if (is_array($v[0])) {
+                                foreach ($v as $field) {
+                                    $field_name = $field['field_name'];
+                                    $field_value = $field['field_value'];
+                                    if (!isset($field_name) || !isset($field_value) ||
+                                        !UserManager::update_extra_field_value($userId, $field_name, $field_value)) {
+                                        throw new Exception(get_lang('CouldNotUpdateExtraFieldValue').': '.print_r($field, true));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line exceeds maximum limit of 120 characters; contains 130 characters](https://app.codacy.com/app/chamilo/chamilo-lms/pullRequest?prid=4554450)